contracts/truefi2/TrueCreditAgency.sol

```diff
@@ -345,6 +350,11 @@ contract TrueCreditAgency is UpgradeableClaimable, ITrueCreditAgency {
         (uint8 oldScore, uint8 newScore) = _updateCreditScore(pool, msg.sender);
         require(newScore >= minCreditScore, "TrueCreditAgency: Borrower has credit score below minimum");
         require(amount <= borrowLimit(pool, msg.sender), "TrueCreditAgency: Borrow amount cannot exceed borrow limit");
+
+        if(totalBorrowed(msg.sender, 18) == 0) {
+            borrowingMutex.lock(msg.sender, address(this));
```

Comment:
This will implicitly fail if the mutex is already locked, which is obviously right, but not sure if we shouldn't check it explicitly.

Reply:
With error a message like `TrueCreditAgency: Borrower cannot open two parallel debt positions`, or something like that.

Reply:
In which way?

Reply:
You can add a similar line https://github.com/trusttoken/smart-contracts/blob/8374c789f31e37b4880979c38b5e1d7cd9853848/contracts/truefi2/TrueLender2.sol#L353